src/search/components/SearchForm.js

```diff
@@ -0,0 +1,30 @@
+import React, { PropTypes } from 'react';
+
+const propTypes = {
+  onSearch: PropTypes.func.isRequired,
+};
+
+class SearchForm extends React.Component {
+  constructor() {
+    super();
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
```

Comment:
I tried that and `eslint` chokes on it. I'm not sure if that used to be ES6 but it is now ES7+ and doesn't appear to have any consensus.

The [airbnb react styleguide](https://github.com/airbnb/javascript/tree/master/react#methods) suggests putting them in the constructor, I'm guessing because that spec isn't complete.